apps/lockfile-explorer-web/src/parsing/lfxGraphSerializer.ts

```diff
@@ -0,0 +1,134 @@
+
+    const jsonLfxEntry: IJsonLfxEntry = {
+      jsonId: nextIndex,
+
+      kind: entry.kind,
+      entryId: entry.entryId,
+      rawEntryId: entry.rawEntryId,
+      packageJsonFolderPath: entry.packageJsonFolderPath,
+      entryPackageName: entry.entryPackageName,
+      displayText: entry.displayText,
+      entryPackageVersion: entry.entryPackageVersion,
+      entrySuffix: entry.entrySuffix,
+
+      // Lists will be added in the second loop
+      dependencies: [],
+      transitivePeerDependencies: [],
+      referrerJsonIds: []
+    };
+
+    jsonLfxEntries.push(jsonLfxEntry);
+    jsonIdByEntry.set(entry, jsonLfxEntry.jsonId);
+  }
+
+  // Now that we built the jsonId lookup, we can serialize the lists.
+  for (let i: number = 0; i < jsonLfxEntries.length; ++i) {
+    const jsonLfxEntry: IJsonLfxEntry = jsonLfxEntries[i];
+    const entry: LockfileEntry = graph.entries[i];
+
+    for (const dependency of entry.dependencies) {
+      const jsonLfxDependency: IJsonLfxDependency = {
+        name: dependency.name,
+        version: dependency.version,
+        entryId: dependency.entryId,
+        dependencyType: dependency.dependencyType,
+        peerDependencyMeta: {
+          name: dependency.peerDependencyMeta.name,
+          version: dependency.peerDependencyMeta.version,
+          optional: dependency.peerDependencyMeta.optional
+        }
+      };
+      if (dependency.resolvedEntry) {
+        jsonLfxDependency.resolvedEntryJsonId = toJsonId(dependency.resolvedEntry);
+      }
+
+      jsonLfxEntry.dependencies.push(jsonLfxDependency);
+    }
+    jsonLfxEntry.transitivePeerDependencies = Array.from(entry.transitivePeerDependencies);
+    jsonLfxEntry.referrerJsonIds = entry.referrers.map((x) => toJsonId(x));
+  }
+
+  return { entries: jsonLfxEntries };
+}
+
+export function deserializeFromJson(jsonLfxGraph: IJsonLfxGraph): LfxGraph {
+  const graph: LfxGraph = new LfxGraph();
+
+  const entries: LockfileEntry[] = graph.entries;
+
+  function fromJsonId(jsonId: number): LockfileEntry {
+    const result: LockfileEntry | undefined = entries[jsonId];
+    if (result === undefined) {
+      throw new Error('Invalid jsonId');
+    }
+    return result;
+  }
+
+  const jsonLfxEntries: IJsonLfxEntry[] = jsonLfxGraph.entries;
+
+  // First create empty entries
+  for (const jsonLfxEntry of jsonLfxEntries) {
+    entries.push(new LockfileEntry(jsonLfxEntry.kind));
+  }
+  for (let i: number = 0; i < jsonLfxEntries.length; ++i) {
+    const jsonLfxEntry: IJsonLfxEntry = jsonLfxEntries[i];
+    const entry: LockfileEntry = graph.entries[i];
+
+    entry.entryId = jsonLfxEntry.entryId;
+    entry.rawEntryId = jsonLfxEntry.rawEntryId;
+    entry.packageJsonFolderPath = jsonLfxEntry.packageJsonFolderPath;
+    entry.entryPackageName = jsonLfxEntry.entryPackageName;
+    entry.displayText = jsonLfxEntry.displayText;
+    entry.entryPackageVersion = jsonLfxEntry.entryPackageVersion;
+    entry.entrySuffix = jsonLfxEntry.entrySuffix;
```

Comment:
```suggestion
    Object.assign(entry, jsonLfxEntry);
```

Reply:
This is deserialization, not copying.  Each field needs to be mapped carefully.